libnmstate/nm/checkpoint.py

```diff
@@ -92,21 +49,97 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     def create(self):
         devs = []
         timeout = self._timeout
-        cp_flags = (
-            CHECKPOINT_CREATE_FLAG_DESTROY_ALL |
-            CHECKPOINT_CREATE_FLAG_DELETE_NEW_CONNECTIONS |
-            CHECKPOINT_CREATE_FLAG_DISCONNECT_NEW_DEVICES
-        )
-        cp = self._manager.interface.CheckpointCreate(devs, timeout, cp_flags)
-        logging.debug('Checkpoint %s created for all devices: %s', cp, timeout)
-        self._checkpoint = cp
+        cp_flags = (nmclient.NM.CheckpointCreateFlags.DESTROY_ALL |
+                    nmclient.NM.CheckpointCreateFlags.DELETE_NEW_CONNECTIONS |
+                    nmclient.NM.CheckpointCreateFlags.DISCONNECT_NEW_DEVICES)
+        userdata = self
+
+        self._mainloop.push_action(
+            self._client.checkpoint_create,
+            devs, timeout, cp_flags, None, _checkpoint_create_callback,
+            userdata)
+
+        self._mainloop.run(timeout)
 
     def destroy(self):
-        self._manager.interface.CheckpointDestroy(self._checkpoint)
-        logging.debug('Checkpoint %s destroyed', self._checkpoint)
+        if self._checkpoint:
+            userdata = self
+            self._mainloop.push_action(
+                self._client.checkpoint_destroy,
+                self._checkpoint.get_path(), None,
+                _checkpoint_destroy_callback, userdata)
+            self._mainloop.run(self._timeout)
+        else:
+            logging.warning('No checkpoint to destroy')
 
     def rollback(self):
-        result = self._manager.interface.CheckpointRollback(self._checkpoint)
-        logging.debug('Checkpoint %s rollback executed: %s',
-                      self._checkpoint, result)
-        return result
+        if self._checkpoint:
+            userdata = self
+            self._mainloop.push_action(
+                self._client.checkpoint_rollback,
+                self._checkpoint.get_path(), None,
```

Comment:
One per line please and place the `None` in a descriptive variable so we will know what it represents.

Reply:
I changed the `None` to `self._mainloop.cancellable`.

And done for the one per line.